src/PageLayout/PageLayout.tsx

```diff
@@ -563,11 +566,16 @@ const Pane = React.forwardRef<HTMLDivElement, React.PropsWithChildren<PageLayout
     },
     forwardRef,
   ) => {
-    if (responsivePosition !== undefined) {
-      // eslint-disable-next-line no-console
-      console.warn(
-        'The `position` prop will be removed on the next major version. You should order your markup as you want it to render instead.',
-      )
+    if (__DEV__ && process.env.NODE_ENV !== 'test') {
+      // We don't want these warnings to show up on tests because it fails the tests (at dotcom) due to not extecting a warning.
+      // Practically, this is not a conditional hook, it is just making sure this hook runs only on DEV not PROD.
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      React.useEffect(() => {
```

Comment:
Do we still need to wrap it in a useEffect if it's  in `__DEV__ && process.env.NODE_ENV !== 'test'`?

Reply:
Hmmm, I imagine keeping the implementation same in prod and dev would help identify issues easier in general but it might not be needed. What do _you_ think?

The reason I added the useEffect to render the warning only when the dependency `responsivePosition` is changed rather than every render but happy to remove.